plugins/ibm_qradar/icon_ibm_qradar/util/utils.py

```diff
@@ -0,0 +1,138 @@
+import json
+import logging
+
+from insightconnect_plugin_runtime.exceptions import (
+    ConnectionTestException,
+    PluginException,
+    ClientException,
+)
+from requests import Response
+
+from icon_ibm_qradar.util.constants.constant import (
+    SUCCESS_RESPONSE_CODE,
+)
+from icon_ibm_qradar.util.constants.messages import INVALID_RANGE_FOUND
+from icon_ibm_qradar.util.url import URL
+
+
+def prepare_request_params(
+    params: {},
+    logger: logging.Logger,
+    url_obj: URL,
+    input_fields: list,
+    query_params: dict = None,
+):
+    """Parse the common input provided including range, sort, filter and fields.
+
+    :param params: Input Dict
+    :param logger: Logging logger
+    :param url_obj: URL object
+    :param input_fields: Possible input fields for the action
+    :param query_params: pre set query params
+    :return:
+    """
+    if not query_params:
+        query_params = {}
+
+    headers = get_default_header()
+    basic_url = url_obj.get_basic_url()
+
+    for field in input_fields:
+        query_value = params.get(field, "")
+
+        if field == "range" and query_value != "":
+            check, query_range = validate_query_range(query_value)
+            if not check:
+                logger.error(query_range)
+                logger.info("Terminating: query_range provided is invalid")
+                raise ClientException(Exception(INVALID_RANGE_FOUND))
+            headers["Range"] = f"items={query_range}"
+        else:
+            if query_value != "":
+                query_params[field] = query_value
+
+    final_queries = []
+    for key in query_params:
+        final_queries.append(f"{key}={query_params[key]}")
+
+    if len(final_queries) > 0:
+        basic_url = f"{basic_url}?{'&'.join(final_queries)}"
+
+    return basic_url, headers
+
+
+def validate_query_range(query_range) -> (bool, str):
+    """To validate the provided range for the output list.
+
+    :param query_range: string representation of range
+    :return: boolean
+    """
+    try:
+        start, end = query_range.split("-")
+        start = int(start.strip())
+        end = int(end.strip())
+        if start < 0:
+            return False, "Range should start from 0 or greater than 0 "
+        if end < 0:
+            return False, "Range should start end 0 or greater than 0"
+
+        return True, f"{start}-{end}"
+
+    except ValueError as err:
+        return False, err
+
+
+def get_default_header() -> dict:
+    """To get the default headers."""
+    return {"Accept": "application/json"}
+
+
+def delete_none(_dict):
+    """To remove the key having none value in the dict.
+
+    :param _dict: input dict.
+    :return:
+    """
+    if isinstance(_dict, dict):
```

Comment:
```suggestion
def delete_none(dict_):
    """To remove the key having none value in the dict.

    :param _dict: input dict.
    :return:
    """
    if isinstance(dict_, dict):
```
To conform with PEP8